### PR TITLE
Track shard-level stats

### PR DESCRIPTION
**Pull request by otoolep**

### CODE REVIEW

**Comment by otoolep** [comment]
```
$ curl -G http://localhost:8086/query?pretty=true --data-urlencode "db=internal" --data-urlencode "q=SHOW STATS"
{
    "results": [
        {
            "series": [
                {
                    "name": "server",
                    "columns": [
                        "shardsCreated",
                        "writeSeriesMessageTx",
                        "broadcastMessageTx",
                        "broadcastMessageRx",
                        "batchWriteRx",
                        "pointWriteRx"
                    ],
                    "values": [
                        [
                            1
                        ],
                        [
                            1
                        ],
                        [
                            4
                        ],
                        [
                            4
                        ],
                        [
                            1
                        ],
                        [
                            1
                        ]
                    ]
                },
                {
                    "name": "shard1",
                    "columns": [
                        "writeSeriesMessageRx",
                        "shardBytes",
                        "shardWrite"
                    ],
                    "values": [
                        [
                            1
                        ],
                        [
                            26
                        ],
                        [
                            1
                        ]
                    ]
                }
            ]
        }
    ]
}
```

**Comment by otoolep** [comment]
Looking at build failure.

**Comment by otoolep** [comment]
Green build. Unrelated test failure, filed as ticket https://github.com/influxdb/influxdb/issues/2009

File: server.go [hunk 1]
```diff
@@ -328,22 +328,40 @@ func (s *Server) StartSelfMonitoring(database, retention string, interval time.D
...
+	// server's Raft ID.
+	pointFromStats := func(st *Stats) Point {
+		point := Point{
+			Timestamp: time.Now(),
+			Name:      st.Name(),
+			Tags:      map[string]string{"raftID": strconv.FormatUint(s.id, 10)},
+			Fields:    make(map[string]interface{}),
+		}
+		st.Walk(func(k string, v int64) {
+			point.Fields[k] = int(v)
```
Comment (rothrock) [inline server.go]: just curious, why int(v)? isn't v already an int?

**Comment by rothrock** [comment]
Looks good to me.

File: server.go (hunk 1 above)
Comment (otoolep) [inline server.go]: I guess int64 != int

**Comment by sanga** [comment]
isn't db=internal redundant here btw? Seems like an implementation detail that the user needn't care about.

**Comment by otoolep** [comment]
Users need to know the database so they can query the stats via InfluxQL.

**Pull request merged**